sdk/src/test/java/com/google/cloud/dataflow/sdk/transforms/DoFnTest.java

```diff
@@ -188,4 +193,30 @@ public void processElement(DoFn<String, String>.ProcessContext c) throws Excepti
 
     return pipeline;
   }
+  
+  @Test
+  public void testPopulateDisplayDataDefaultBehavior() {
+    DoFn<String, String> usesDefault =
+        new DoFn<String, String>() {
+          @Override
+          public void processElement(DoFn<String, String>.ProcessContext c) throws Exception {}
+        };
+
+    DisplayData data = collectDisplayData(usesDefault);
+    assertThat(data.items(), empty());
+  }
+
+  private DisplayData collectDisplayData(final DoFn<String, String> fn) {
```

Comment:
It might be useful to add a `DisplayData.from(HasDisplayData root)`, which just returns the display data starting from the given root. Then we don't need this special case, and we can use that in general for testing a lot of components.